src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
Guess the only way to avoid that (us sending [many] requests fast and remote taking long, so everything arrives in the async_wait) is to have some sort of window.

E.g. with window=100, call_many would only send requests while the "responses we are waiting for" list length is below 100. Otherwise it would continue looping waiting for a response.